sktime/benchmarking/_lib_mini_kotsu/run.py

```diff
@@ -98,11 +98,63 @@ def run(
 
             model = model_spec.make()
             results, elapsed_secs = _run_validation_model(validation, model, run_params)
+            results, y_pred, y_test = results
```

Comment:
here, we are changing `results` - does this not lead to modified behaviour below?

Reply:
Yes, this is the reason why I needed to adapt also the tests. 

The alternative to put this into `results` was a bit ugly programming wise in the `write_results` methods. Thus, I used this approach. However, if we make more architectural changes as proposed by you, this will change again.

Reply:
is this changing any of the public interfaces?